cookbooks/fb_nginx/resources/create_certs.rb

```diff
@@ -0,0 +1,46 @@
+#
+# Cookbook:: fb_nginx
+# Recipe:: default
+#
+# Copyright (c) 2019-present, Vicarious, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+action :enable do
+  node['fb_nginx']['sites'].each do |site, config|
+    if !config['_create_self_signed_cert'] || !config['ssl_certificate'] ||
+       !config['ssl_certificate_key']
+      Chef::Log.debug(
+        "fb_stunnel[#{site}]: Not creating cert: missing " +
+        '"ssl_certificate", "ssl_certificate_key", or ' +
+        '"_create_self_signed_cert"',
+      )
+      next
+    end
+    if ::File.exist?(config['ssl_certificate'])
+      Chef::Log.debug(
+        "fb_stunnel[#{site}]: Not creating cert: it already exists.",
+      )
+      next
+    end
+
+    execute "create key/cert for #{site}" do
+      command 'openssl req -new -newkey rsa:4096 -x509 -sha256 -days 365 ' +
```

Comment:
This will fail if the openssl CLI is missing (same as stunnel)

Reply:
I'm adding an error but not breaking the run (the run will break when the service can't start). I'm not sure which is better....